app/controllers/product_entries_controller.rb

```diff
@@ -5,7 +5,9 @@ class ProductEntriesController < AuthenticationController
   before_action :file_uploaded?, only: %i[create_by_csv]
 
   def index
-    @product_entry = ProductEntry.all
+    return @product_entry = ProductEntry.all if current_user.admin?
+
+    @product_entry = current_user.warehouse.product_entries
```

Comment:
Não sei se você vai gostar dessa forma mas é legal saber que existe hahaha, eu particularmente acho que prefiro a que você usou. Mas essa aqui é fancy:

```suggestion
    @product_entry = if current_user.admin?
                                    ProductEntry.all
                                   else
                                     current_user.warehouse.product_entries
                                   end
```